addons/account/models/account_report.py

```diff
@@ -280,6 +280,14 @@ def _validate_formula(self):
                     expression.report_line_id.display_name,
                 ))
 
+    @api.constrains('sequence')
+    def _validate_account_report_line(self):
+        parent_set = set()
+        for line in self.report_id.line_ids.sorted('sequence'):
```

Comment:
You need to sort using the `id` as well[^1], and ignore when the sequence is not set (the sequence widget should write the values one by one on all lines the first time)
```suggestion
        for line in self.report_id.line_ids.sorted(lambda l: (l.sequence, l.id)).filtered('sequence'):
```
Otherwise, 

[^1]: note that all the sequences are 0 by default -- for stupid reasons, but that's the case anyways.